### DISCUSSION TITLE
fastdds spy tool support new idl topic type

### DISCUSSION THREAD

**Discussion by wannastar** [eProsima/Fast-DDS-spy#51] [Q&A]
fastdds spy tool only support one topic type - ShapeType. However, i wanna to support my self-defined type to support show command.(fastdds_spy show <mytopic>)

**Comment by rsanchez15** [comment]
Hi @wannastar,

Fast DDS Spy tool does not know any type but discovers them dynamically using dynamic types. That is, the ``ShapeType`` type you are looking at is not pre-loaded or pre-compiled by the tool, simply the Shapes Demo application announces its type so that other applications can discover it. In this way, the Spy discovers the ``ShapeType`` type and from there it can handle the data coming from this type. 

I recommend you to go to [this tutorial](https://dds-recorder.readthedocs.io/en/latest/rst/tutorials/dynamic_types.html) on how to configure and enable Dynamic Types in your Fast DDS application so Fast DDS Spy can discover it.
